server/services/__tests__/offenderService.spec.js

```diff
@@ -340,7 +340,7 @@ describe('Offender Service', () => {
 
     it('should return an error when no booking Id is passed', async () => {
       const repository = {
-        getEventsFor: jest.fn().mockReturnValue([]),
+        getEventsFor: jest.fn().mockReturnValue({ error: true }),
```

Comment:
very meh... guess this should be resolved rather than return as this will return a promise

Reply:
So Async automatically wraps any response in a Promise I think. There are 21 other instances of _ mockReturnValue_ in this test block where the function that is mocked is _async_. From what I can see all the _real_ functions end up being auto wrapped, so I guess that was the reasoning in the existing tests? I guess returning a Promise would ensure that we were unwrapping with an _await_ or _then_ though? 
I'm easy either way 🤷

Reply:
yes basically - its the kind of thing that typescript would have detected.

I have seen (done) bugs where people (me) have forgotten to await promises and tests have passed - but here you need to remember to mockResolve so its kind of a double accounting